sdk/storage/storage-blob/review/storage-blob.api.md

```diff
@@ -1041,6 +1058,7 @@ export class BlobServiceClient extends StorageClient {
     deleteContainer(containerName: string, options?: ContainerDeleteMethodOptions): Promise<ContainerDeleteResponse>;
     findBlobsByTags(tagFilterSqlExpression: string, options?: ServiceFindBlobByTagsOptions): PagedAsyncIterableIterator<FilterBlobItem, ServiceFindBlobsByTagsSegmentResponse>;
     static fromConnectionString(connectionString: string, options?: StoragePipelineOptions): BlobServiceClient;
+    generateAccountSasUrl(permissions: AccountSASPermissions, expiresOn: Date, resourceTypes: string, options?: ServiceGenerateAccountSasUrlOptions): string;
```

Comment:
Thoughts on making this `generateSasUrl` like the others? Or thoughts on making the others e.g. `generateContainerSasUrl` etc.?

Reply:
can resourceTypes be defaulted?

Reply:
>Thoughts on making this generateSasUrl like the others? Or thoughts on making the others e.g. generateContainerSasUrl etc.?

Following .NET for the naming. They haven't released it yet. We can argue it and ask other languages to align if you have a strong preference.

> can resourceTypes be defaulted?

I think it's a good idea to default these params. How about we also default `permission` to `r` (Read, useful for Blob and container) and `expiresOn` to a hour later (not so sure about this one, probably should put it in the first place in the params list)?

Reply:
Note that account-level SAS and service-level SAS are two types of SAS. I wonder if they named them with this consideration.
https://docs.microsoft.com/en-us/rest/api/storageservices/delegate-access-with-shared-access-signature#types-of-shared-access-signatures

Reply:
I'm good with your suggested defaults. 

I see what you mean about account SAS being different in shape (though similar in purpose), so I'm fine giving them more distinctive naming. I think that otherwise there is an easy cognitive mistake to equate "a SAS is just a key to a thing, sometimes an account, sometimes a specific service entity" when the features of each kind of SAS are not actually equivalent.

Reply:
I like the defaults too!

I guess while I see the point about the SAS being different, but does that argue for the others to be `generateServiceSasUrl`?

Reply:
Return `Promise<string>`.

Reply:
Won't return `Promise<string>` for File and Queue service SAS interface as user delegation SAS is supported for the Blob service only.